stringbuilder.go

```diff
@@ -56,6 +59,24 @@ func (s *StringBuilder) AppendRune(char rune) *StringBuilder {
 	return s
 }
 
+// Appends a single integer to the StringBuilder instance
+func (s *StringBuilder) AppendInt(integer int) *StringBuilder {
+	return s.Append(strconv.Itoa(integer))
+}
+
+// Appends a single boolean to the StringBuilder instance
+func (s *StringBuilder) AppendBoolean(flag bool) *StringBuilder {
+	return s.Append(strconv.FormatBool(flag))
+}
+
+// Appends a list of strings to the StringBuilder instance
+func (s *StringBuilder) AppendList(words []string) *StringBuilder {
+	for _, word := range words {
```

Comment:
I was wondering if it makes sense to check beforehand the size of new characters that would go into the StringBuilder and resize only once.
If we have a big list, we might resize it multiple times.

Reply:
Added a `resize` method that is called from both the `Append` & `AppendList` method. In the scenario of a large list, the method call from `AppendList` will resize it with total length of all the words. Post that calling `Append` for each individual word will not invoke the `grow` function as the length is already updated.

Reply:
Like the way you solved it!